chain/lf3/ec.go

```diff
@@ -20,52 +20,55 @@ import (
 	"github.com/filecoin-project/lotus/chain/vm"
 )
 
+var (
+	_ ec.Backend = (*ecWrapper)(nil)
+	_ ec.TipSet  = (*f3TipSet)(nil)
+
+	emptyBeacon = make([]byte, 32)
+)
+
 type ecWrapper struct {
 	ChainStore   *store.ChainStore
 	StateManager *stmgr.StateManager
 }
 
-var _ ec.TipSet = (*f3TipSet)(nil)
-
-type f3TipSet types.TipSet
-
-func (ts *f3TipSet) cast() *types.TipSet {
-	return (*types.TipSet)(ts)
+type f3TipSet struct {
+	*types.TipSet
```

Comment:
I don't love the pointer chasing but 🤷 doesn't matter much.

Reply:
Happy to revert it. I thought it reads better than casting but that's entirely subjective on my part.

Reply:
it reads better for sure